src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs

```diff
@@ -302,58 +308,132 @@ public static int LongestCommonPrefixLength(List<string> strings)
+        public Task ProcSelfStatus()
         {
-            public long Size { get; set; }
-            public long Count { get; set; }
+            using (var currentProcess = Process.GetCurrentProcess())
+            {
+                var path = $"/proc/{currentProcess.Id}/status";
+                var txt = File.ReadAllLines(path);
+                using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+                {
+                    var djv = new DynamicJsonValue();
+                    var djvHumanly = new DynamicJsonValue();
+                    foreach (var line in txt)
+                    {
+                        var parsedLine = line.Split(new[] {' ', '\t'}, StringSplitOptions.RemoveEmptyEntries);
+
+                        if (parsedLine.Length < 2) // format may vary
+                            continue; // an error but we ignore and do not put in the results
+
+                        var key = parsedLine[0].Split(":")[0];
+
+                        if (parsedLine.Length == 3 && 
+                            ( parsedLine[2].Equals("kB", StringComparison.InvariantCultureIgnoreCase) ||
+                              parsedLine[2].Equals("mB", StringComparison.InvariantCultureIgnoreCase) ||
+                              parsedLine[2].Equals("gB", StringComparison.InvariantCultureIgnoreCase)))
+                        {
+                            if (long.TryParse(parsedLine[1], out var result) == false)
+                            {
+                                djv[key] = "Failed To Parse Value : " + parsedLine[1];
+                                continue;
+                            }
+
+
+                            switch (parsedLine[2].ToLowerInvariant())
+                            {
+                                case "kb":
+                                    result *= 1024L;
+                                    break;
+                                case "mb":
+                                    result *= 1024L * 1024;
+                                    break;
+                                case "gb":
+                                    result *= 1024L * 1024 * 1024;
+                                    break;
+                            }
+                            djv[key] = result;
+                            djvHumanly[key] = Size.Humane(result);
+                        }
+                        else
+                        {
+                            var restOfTheLine = new StringBuilder(parsedLine[1]);
+                            for (int i = 2; i < parsedLine.Length; i++)
+                            {
+                                restOfTheLine.Append(" ");
+                                restOfTheLine.Append(parsedLine[i]);
+                            }
+                            djv[key] = restOfTheLine.ToString();
+                        }
+                    }
```

Comment:
The same comment as above. Very fragile code. Let's consider to output raw lines as well.